rules/Collection22/Rector/CriteriaOrderingConstantsDeprecationRector.php

```diff
@@ -0,0 +1,217 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Doctrine\Collection22\Rector;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\Array_;
+use PhpParser\Node\Expr\ArrayItem;
+use PhpParser\Node\Scalar\String_;
+use PHPStan\Type\ObjectType;
+use Rector\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see
+ */
+final class CriteriaOrderingConstantsDeprecationRector extends AbstractRector
+{
+    private ObjectType $criteriaObjectType;
+
+    public function __construct()
+    {
+        $this->criteriaObjectType = new ObjectType('Doctrine\Common\Collections\Criteria');
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Replace ASC/DESC with enum Ordering in Criteria::orderBy method call, and remove usage of Criteria::ASC and Criteria::DESC constants elsewhere',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+                    <?php
```

Comment:
php open tag seems cause use statement prefixed in code example, and become:

```

namespace RectorPrefix202408;
use RectorPrefix202408\Doctrine\Common\Collections\Criteria;
```

ref https://github.com/rectorphp/rector/commit/b8c4f153f83386ec0da64a2c275564743b4eb438#diff-c260cb91d355f253688b74213c5bb7c75414832d6c18ba756b6ba3accf08964f

Reply:
https://github.com/rectorphp/rector-doctrine/pull/341